WSL Update

Baka-Noob opened the pull request:
- Adicionadas instruções detalhadas para instalação do WSL (Windows Subsystem for Linux).
- Incluída orientação para instalação de ferramentas essenciais de compilação: arm-none-eabi-gcc, git, make, cmake e g++.
- Ajustes no PATH para integração dos arquivos do STM32, otimizando o ambiente de desenvolvimento.
- Melhorias gerais na clareza e estrutura do guia.

jairchtq committed:
🔧 Update cubemx windows installation

jairchtq committed:
🔧 Update CubeMX installation & add images

Baka-Noob committed:
Implementando o WSL

Consta nesta atualização do Guide, a adição de novas orientações de instalação dos recursos Make, CMake, GIT e GCC através do WSL.

Adicionado também, orientações para a instalação do WSL propriamente dito, assim como a virtualização da BIOS/UEFI que se faz necessária para a instalação do WSL.

Baka-Noob committed:
Update README.md

Acrescentado o comando para a instalação do G++

Adotado um modelo exemplo de união dos comandos "Instalação no Windows" com "Instalação no Linux", já que os comandos são os mesmos. Opinem qual forma preferem

Baka-Noob committed:
Atualização definitiva WSL/Linux

Consta neste commit o acréscimo de informações referentes as instruções de instalação de recursos para compilação em terminal Ubuntu.

Realizadas mudanças na formatação dos tópicos.

Adicionadas melhorias nas informações.

Baka-Noob committed:
hotfix

Pequena correção de sintaxe

Baka-Noob committed:
rotfix 2

pequena alteração de sintaxe

jairchtq committed:
🔧 Update CubeMX and Programmer

jairchtq committed:
Merge branch 'update/windows_wsl' of github.com:ThundeRatz/STM32Guide into update/windows_wsl

jairchtq committed:
📝 Update CubeMX and Programmer installation on WSL

PedroDeSanti commented on README.md:
```diff
@@ -11,14 +11,15 @@
-    - [Instalação no Linux](#instalação-no-linux-1)
+    - [Instalação WSL/Linux](#instalação-WSL/Linux-1)
+  - [G++](#G++)
+    - [Instalação WSL/Linux](#instalação-WSL/Linux-2) 
```
```suggestion
    - [Instalação WSL/Linux](#instalação-wsllinux)
  - [G++](#G++)
    - [Instalação WSL/Linux](#instalação-wsllinux-1) 
```

PedroDeSanti replied:
do jeito que tava nao vai funcionar

PedroDeSanti replied:
alias acho q como só tem uma uma forma de instalar pra cada ferramenta dava até pra tirar esse subtitulo ai

PedroDeSanti commented:
Tem uns probleminhas no sumário pra arrumar recomendo usar a extensao do vscode de markdown q ela já atualiza os índices automaticamente https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one

tem essa outra tbm q é util:
https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint

PedroDeSanti commented on README.md:
```diff
@@ -1606,3 +1677,57 @@ diferenças nas saídas).
+![virtualização ON](https://github.com/user-attachments/assets/4109eded-9a59-48f6-bf46-1b09f19a6374)
+
+Se já está habilitado, prossiga para o passo 1 da [instalação do WSL](#Instalando-o-WSL). 
+Caso contrário, acesse este [guia](#https://support.microsoft.com/pt-br/windows/ativar-a-virtualiza%C3%A7%C3%A3o-no-windows-c5578302-6e43-4b4b-a449-8ced115f58e1) da Microsoft pelo seu celular e siga o passo a passo no seu computador.
```
```suggestion
Caso contrário, acesse este [guia](https://support.microsoft.com/pt-br/windows/ativar-a-virtualiza%C3%A7%C3%A3o-no-windows-c5578302-6e43-4b4b-a449-8ced115f58e1) da Microsoft pelo seu celular e siga o passo a passo no seu computador.
```

PedroDeSanti commented on README.md:
```diff
@@ -1606,3 +1677,57 @@ diferenças nas saídas).
+    
+:::warning
+:warning: Será necessário fazer alterações na sua UEFI/BIOS. Caso não esteja acostumado com isso, peça ajuda aos veteranos.
+:::    
```
```suggestion
> [!WARNING]
> Será necessário fazer alterações na sua UEFI/BIOS. Caso não esteja acostumado com isso, peça ajuda aos veteranos.
```

Não tenho ctz mas acho que vc queria fazer isso:

> [!WARNING]
> Será necessário fazer alterações na sua UEFI/BIOS. Caso não esteja acostumado com isso, peça ajuda aos veteranos.

PedroDeSanti commented on README.md:
```diff
@@ -84,15 +88,24 @@ específicas estarão em suas respectivas seções.
 
-Extraia o .zip e execute o arquivo SetupSTM32CubeMX-X.X.X.exe (X.X.X é a
-versão). Siga as instruções da tela.
+Vamos usar a versão 6.13 do CubeMX. Clique em "Get Software" e selecione a versão 6.13 em "Select version".
+
+![Cube windows installation](media/cubemx_download.png)
```
Algum motivo específico pra escolher essa versão?Acho melhor não deixar a versão do cube pq isso vai deixar o guia desatualizado mais rápido

jairchtq replied:
É que tentando compilar algum projeto, ele dava erro, mesmo gerando o cube direto pelo CubeMX, mas testando na versão 6.13 ele não dava mais erro. Eu tinha mandado no grupo da Comp e mais pessoas tinham falado que não conseguiam

PedroDeSanti commented on README.md:
```diff
@@ -84,15 +88,24 @@ específicas estarão em suas respectivas seções.
+
+![Cube windows installation](media/cubemx_download.png)
+
+Antes de poder baixar o software é necessário estar logado em uma conta da ST. Caso não tenha, crie uma antes de continuar.
+
```
da pra baixar como guest, entao não sei se isso é uma infromação tao relevante pro guia

jairchtq replied:
Para gerar o cube pelo CubeMX pela primeira vez ou para atualizar os arquivos do CubeMX, acho que ele pede para logar antes de continuar, aí acho que seria bom criar uma conta já na hora da instalação

PedroDeSanti commented on README.md:
`````diff
@@ -154,79 +167,137 @@ Categories=Development;Electronics;
+
+````bash
+g++ arquivo_main.cpp arquivo_funções.cpp -o nome_do_executavel.exe
+````
`````
ué mas pq tem o g++ de dependencia?

PedroDeSanti replied:
a gente usa arm-none-eabi-gcc pra compilar o os nossos projetos

PedroDeSanti commented on README.md:
````diff
@@ -154,79 +167,137 @@ Categories=Development;Electronics;
+$ git --version
 ```
 
+
````
Sinceramente acho q da pra resumir tudo isso num topico só, falando que as ferramentas necessárias buildar o projeto sao make cmake e armgcc, explicar pra q serve cada uma e só botar que pra instalar no ubuntu roda

```bash
sudo apt install cmake make gcc-arm-none-eabi
```

e depois a parte do git explica que se quiser versionar codigo e usar nosso template

```bash
sudo apt install git
```

PedroDeSanti commented on README.md:
```diff
@@ -154,79 +167,137 @@ Categories=Development;Electronics;
+
+![Cube programmer path](media/cube_programmer_path.png)
+
+Após isso, no terminal do WSL digite `source ~/.bashrc` e em seguida `STM32_Programmer_CLI.exe -l`, se tudo foi configurado corretamente, o comando vai retornar a versão do Cube Programmer e outras coisas.
```
Como isso (forma de adicionar um caminho do windows em uma variavel de ambiente do WSL) ta repetindo, tem tanto aqui quanto na instalcão do cube), acho que vale passar isso pra dentro do apêndice "Colocando caminhos no PATH", dai da pra adicionar um subtopico pro wsl

PedroDeSanti replied:
inclusive, se agora só vai usar wsl pro desenvolvimento, acho que da pra tirar a parte de colocar caminho no path do windows, pq nao ta sendo mais usado

PedroDeSanti replied:
dai da pra deixar a parte de colocando caminhos no path do linux, e depois pro wsl é a mesma coisa, mas com a particularidade do /mnt/c/

PedroDeSanti commented on README.md:
```diff
@@ -84,15 +88,24 @@ específicas estarão em suas respectivas seções.
+
 ### Instalação no Linux
 
 É necessário instalar Java. Para isso, execute os seguintes comandos:
```
Acho q já q vc ta mexendo na parte de instalação, dava pra aproveitar e atualizar a instalação do cube no linux, pq nao precisa mais intalar Java

PedroDeSanti commented on README.md:
````diff
@@ -154,79 +167,137 @@ Categories=Development;Electronics;
 
-### Instalação no Linux
+```bash
+$ sudo apt install git 
````
```suggestion
sudo apt install git 
```

Acho que é melhor deixar sem o $, fica mais fácil de copiar. O $ só faz sentido quando vc ta escrevendo um comendo em um output, daí serve pra identificar o que é comando e o que é output. Inclusive o linter de markdown até reclama quando vc coloca $ e não coloca um output

PedroDeSanti replied:
inclusive seria legal se tirasse todos kkkkkkkkkk

PedroDeSanti commented:
desculpa kkk acabei comentando muita coisa kkkk

Baka-Noob committed:
Repaginando os últimos commits

Após ótimas observações do @PedroDeSanti (tmj lek), pude melhorar as alterações. 
 - Agrupado CMake, make e arm-none-eabi-gcc em um só comando e agora pertencem ao tópico "Compiladores".
 - Ajustes feitos nos índices.
 - Melhorias gerais de formatação.
 - Posicionado o tópico WSL para a parte de cima do índice (antes estava no apêndice).

Baka-Noob committed:
hotfix

Pequena correção de sintaxe

jairchtq committed:
📝 Update path configuration on wsl
